utils.py

```diff
@@ -159,3 +159,23 @@ def _compare_endpoints(endpoint1, endpoint2):
     elif 'out_vlan' in endpoint1 or 'in_vlan' in endpoint2:
         return False
     return True
+
+
+def convert_vlan(value):
+    """Auxiliar function to calculate dl_vlan"""
+    if isinstance(value, int):
+        return value, 4095
+    value, mask = map(int, value.split('/'))
+    return value, mask
+
+
+def match_field_dl_vlan(vlan_value, field_flow):
+    """Verify match in dl_vlan"""
+    value, mask = convert_vlan(vlan_value)
+    vlan = value & (mask & 4095)
+    value_flow, mask_flow = convert_vlan(field_flow)
+    vlan_flow = value_flow & (mask_flow & 4095)
+    # untagged cases
+    if vlan == 0 or vlan_flow == 0:
```

Comment:
`match_field_dl_vlan(0, 10)` would return `True`, which isn't right.

`dl_vlan: 0` arg shouldn't match a flow that has `dl_vlan: 10`.

There's still a misunderstanding how you're handling untagged (dl_vlan: 0). 

Before comparing the resulting `vlan == vlan_flow` the line below, we'll need extra checks here.

Reply:
@gretelliz let's try to also simplify this code here, another thing that's making harder to reason and cover the cases is that it's expecting that both `vlan_value` and `field_flow` can have a mask. The input packet arg vlan_value should always be a single int vlan value, whereas `field_flow` from a flow entry can either be a int or str having a mask. Can you refactor this? That way in the API we can restrict `dl_vlan` to always be from 1 to 4095, and then if they need to send an untagged packet let's only allow it without `dl_vlan` and not the int zero (once the validations are in place then we should have the validation guarantees). We don't need to support a range of values in the input, this will also be compatible with vlan range that will take care of splitting the range and send single individual trace entries, and even when we have intermediary dl_vlan ranges, the outgoing vlan should always be a single one (or being poppped) as well, so that should still be compatible.